tests/test_natural_gradient.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import pytest
+import tensorflow as tf
+
+import gpflow
+from gpflow.config import default_float
+from gpflow.optimizers import NaturalGradient
+from gpflow.utilities import set_trainable
+
+
+class Setup:
+    N, M, D = 4, 3, 2
+    likelihood_variance = 0.1
+
+
+@pytest.fixture
+def data():
+    N, D = Setup.N, Setup.D
+    x = tf.random.normal((N, D), dtype=default_float())
+    y = tf.random.normal((N, 1), dtype=default_float())
+    return (x, y)
+
+
+@pytest.fixture
+def inducing_variable():
+    return tf.random.normal((Setup.M, Setup.D), dtype=default_float())
+
+
+@pytest.fixture
+def kernel():
+    return gpflow.kernels.SquaredExponential()
+
+
+@pytest.fixture
+def likelihood():
+    return gpflow.likelihoods.Gaussian(variance=Setup.likelihood_variance)
+
+
+@pytest.fixture
+def gpr_and_vgp(data, kernel, likelihood):
+    vgp = gpflow.models.VGP(data, kernel, likelihood)
+    gpr = gpflow.models.GPR(data, kernel)
+    gpr.likelihood.variance.assign(likelihood.variance)
+    set_trainable(vgp, False)
+    vgp.q_mu.trainable = True
+    vgp.q_sqrt.trainable = True
+    return gpr, vgp
+
+
+@pytest.fixture
+def sgpr_and_svgp(data, inducing_variable, kernel, likelihood):
+    svgp = gpflow.models.SVGP(kernel, likelihood, inducing_variable)
+    sgpr = gpflow.models.SGPR(data, kernel, inducing_variable=inducing_variable)
+    sgpr.likelihood.variance.assign(Setup.likelihood_variance)
+    set_trainable(svgp, False)
+    svgp.q_mu.trainable = True
+    svgp.q_sqrt.trainable = True
+    return sgpr, svgp
+
+
+def assert_gpr_vs_vgp(m1: tf.Module,
+                      m2: tf.Module,
+                      gamma: float = 1.0,
+                      maxiter: int = 1,
+                      xi_transform: gpflow.optimizers.natgrad.XiTransform = None):
```

Comment:
shouldn't that be `Optional[gpflow.optimizers.natgrad.XiTransform]`?

Reply:
Yes, that should be the optional type.